src/internal/modules/invoice-accounts/controllers/contacts.js

```diff
@@ -84,8 +87,49 @@ const postContactCreate = async (request, h) => {
   return h.postRedirectGet(form, urlJoin('/invoice-accounts/create/', regionId, companyId, 'create-contact'));
 };
 
+const getContactSearch = async (request, h) => {
+  const { regionId, companyId } = request.params;
+  // Return the view
+  const { viewData, agent } = dataService.sessionManager(request, regionId, companyId);
+  let selectedContactId = agent ? agent.id : null;
+
+  return h.view('nunjucks/form', {
+    ...request.view,
+    pageTitle: 'Does this contact already exist?',
+    back: `/invoice-accounts/create/${regionId}/${companyId}?redirectPath=${viewData.redirectPath}`,
+    form: sessionForms.get(request, selectExistingContactForm(request, selectedContactId))
+  });
+};
+
+const postContactSearch = async (request, h) => {
+  const { id, filter } = request.payload;
+  const { regionId, companyId } = request.params;
+  const form = forms.handleRequest(selectExistingContactForm(request), request, selectExistingContactSchema);
+  if (form.isValid) {
+    if (id === 'new') {
+      let volatileKey = uuid();
```

Comment:
you could change to const because the key doesn't change

Reply:
Agreed. I will change this